src/maxminddb.c

```diff
@@ -1481,21 +1520,21 @@ LOCAL uint32_t get_ptr_from(uint8_t ctrl, uint8_t const *const ptr,
 {
     uint32_t new_offset;
     switch (ptr_size) {
-    case 0:
-        new_offset = (ctrl & 7) * 256 + ptr[0];
-        break;
     case 1:
-        new_offset = 2048 + (ctrl & 7) * 65536 + ptr[0] * 256 + ptr[1];
+        new_offset = ( (ctrl & 7) << 8) + ptr[0];
```

Comment:
I find bit shifting easier to read in this case as it makes it clear what is going on.